ValheimPlus/GameClasses/Player.cs

```diff
@@ -135,77 +139,83 @@ private static void Postfix(ref Player __instance, ref Vector3 ___m_moveDir, ref
+                    il[i + 1].opcode == OpCodes.Ldarg_1 /* dt */ &&
+                    il[i + 2].opcode == OpCodes.Add)
                 {
-                    __instance.m_foodRegenTimer = 0f;
-                    float num = 0f;
-                    foreach (Player.Food food2 in __instance.m_foods)
-                    {
-                        num += food2.m_item.m_shared.m_foodRegen;
-                    }
-                    if (num > 0f)
-                    {
-                        float num2 = 1f;
-                        __instance.m_seman.ModifyHealthRegen(ref num2);
-                        num *= num2;
-                        Helper.getPlayerCharacter(__instance).Heal(num, true);
-                    }
+                    // We insert after Ldarg_1 (push dt) a call to our function, which computes the modified DT and returns it.
+                    il.Insert(i + 2, new CodeInstruction(OpCodes.Call, method_ComputeModifiedDt));
                 }
             }
 
-            return false;
+            return il.AsEnumerable();
         }
 
-        private static float getModifiedDeltaTime(ref Player __instance, ref float dt)
+        private static float ComputeModifiedDT(float dt)
         {
-            float defaultDeltaTimeTarget = 1f;
-            float newDetalTimeTarget;
+            return dt / Helper.applyModifierValue(1.0f, Configuration.Current.Food.foodDurationMultiplier);
+        }
+    }
+
+    [HarmonyPatch(typeof(Player), nameof(Player.GetTotalFoodValue))]
+    public static class Player_GetTotalFoodValue_Transpiler
+    {
+        private static FieldInfo field_Food_m_health = AccessTools.Field(typeof(Player.Food), nameof(Player.Food.m_health));
+        private static FieldInfo field_Food_m_stamina = AccessTools.Field(typeof(Player.Food), nameof(Player.Food.m_stamina));
+        private static FieldInfo field_Food_m_item = AccessTools.Field(typeof(Player.Food), nameof(Player.Food.m_item));
+        private static FieldInfo field_ItemData_m_shared = AccessTools.Field(typeof(ItemDrop.ItemData), nameof(ItemDrop.ItemData.m_shared));
+        private static FieldInfo field_SharedData_m_food = AccessTools.Field(typeof(ItemDrop.ItemData.SharedData), nameof(ItemDrop.ItemData.SharedData.m_food));
+        private static FieldInfo field_SharedData_m_foodStamina = AccessTools.Field(typeof(ItemDrop.ItemData.SharedData), nameof(ItemDrop.ItemData.SharedData.m_foodStamina));
+
+        /// <summary>
+        /// Replaces loads to the current health/stamina for food with loads to the original health/stamina for food
+        /// inside Player::GetTotalFoodValue. This disables food degradation.
+        /// </summary>
+        [HarmonyTranspiler]
+        public static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions)
+        {
+            if (!Configuration.Current.Food.IsEnabled) return instructions;
+
+            List<CodeInstruction> il = instructions.ToList();
```

Comment:
Do you think this method would be a good candidate for the `yield return` concept that's recommended for transpilers?

Reply:
Yes, probably... subjectively, I'm not a fan of `yield return`. I think it makes the code harder to understand. But it does have a marginal performance benefit. The refactored code would look something like this (may not work!):

```cs
[HarmonyTranspiler]
public static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions)
{
    bool should_modify = Configuration.Current.Food.IsEnabled && Configuration.Current.Food.foodDegradesOverTime;

    foreach (CodeInstruction inst in instructions)
    {
        bool loads_health = inst.LoadsField(field_Food_m_health);
        bool loads_stamina = inst.LoadsField(field_Food_m_stamina);

        if (should_modify && (loads_health || loads_stamina))
        {
            inst.operand = field_Food_m_item;
            yield return inst;
            yield return new CodeInstruction(OpCodes.Ldfld, field_ItemData_m_shared);
            yield return new CodeInstruction(OpCodes.Ldfld, loads_health ? field_SharedData_m_food : field_SharedData_m_foodStamina);
        }
        else
        {
            yield return inst;
        }
    }
}
```

Reply:
Okay yeah I think we'd rather have maintainability than a marginal performance increase  haha we can keep as is